lib/imagec/layer_cache.go

```diff
@@ -0,0 +1,151 @@
+// Copyright 2016 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package imagec
+
+import (
+	"encoding/json"
+	"fmt"
+	"sync"
+
+	log "github.com/Sirupsen/logrus"
+
+	"github.com/vmware/vic/lib/apiservers/engine/backends/kv"
+	"github.com/vmware/vic/lib/apiservers/portlayer/client"
+	"github.com/vmware/vic/pkg/trace"
+)
+
+// LCache is an in-memory cache to account for existing image layers
+// It is used primarily by imagec when coordinating layer downloads
+// The cache is initially hydrated by way of the image cache at startup
+type LCache struct {
+	m sync.RWMutex
+	// Layers maps from layer id to layer metadata
+	Layers map[string]*ImageWithMeta
```

Comment:
I thought about it, but it has docker's `types.V1Image` embedded inside which is why it is named like this to begin with (correct me if I'm wrong @caglar10ur).

Truth be told, I am not opposed to renaming image->layer everywhere it's appropriate. No small feat, though, since those terms are munged all over the place, originating mainly from the fact that the Portlayer refers to layers as images. :)